programs/zstdcli.c

```diff
@@ -409,8 +449,20 @@ static int readSizeTFromCharChecked(const char** stringPtr, size_t* value)
         if (**stringPtr=='i') (*stringPtr)++;
         if (**stringPtr=='B') (*stringPtr)++;
     }
-    *value = result;
-    return 0;
+
+    //Only well formed entries will be 0 or , (specifically for zstd compound)
+    if (**stringPtr == 0 || **stringPtr == ',') {
+
+        *value = result;
+        return 0;
+    
+    }else{
```

Comment:
This makes sense but I'm not 100% sure.
@felixhandte  / @terrelln would appreciate your inputs here.